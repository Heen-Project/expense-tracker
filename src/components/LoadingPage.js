import React from 'react'

const LoadingPage = () => (
    <div className="loader">
        <img src="/img/loader.gif" className="loader__image" />
    </div>
)

export default LoadingPage