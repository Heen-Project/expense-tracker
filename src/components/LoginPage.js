import React from 'react'
import { connect } from 'react-redux'
import { startLogin } from '../actions/auth'

export const LoginPage = ({ startLogin }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Expense Tracker</h1>
            <p>Let's monitor your expenses from today!</p>
            <button onClick={startLogin} className="button button--login">
                <img src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg" className="button--login-google-sign-in-img"/> Log in with Google
            </button>
            <p style={{fontSize:9, fontWeight:300, marginTop:'10px'}}>Please enable third-party cookies to use the app</p>
        </div>
    </div>
)

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
})

export default connect(undefined, mapDispatchToProps)(LoginPage)