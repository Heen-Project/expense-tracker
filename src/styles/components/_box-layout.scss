.box-layout {
    background: url('/img/bg.webp');
    background-size: cover;
    height: 100vh;
    width: 100vw;
    display: flex;
    justify-content: center;
    align-items: center;
    color: $white;
}

.box-layout__box {
    background: fade-out($grey, 0.15);
    border-radius: 25px;
    text-align: center;
    width: 35rem;
    padding: $l-size $m-size;
}

.box-layout__title {
    margin: 0 0 $m-size 0;
    line-height: 1;
}