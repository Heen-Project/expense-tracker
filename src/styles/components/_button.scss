.button {
    background: $google-green;
    color: $off-white;
    border: none;
    border-radius: 5px;
    font-weight: 300;
    font-size: $font-size-large;
    padding: $s-size $m-size;
    display: inline-block;
    text-decoration: none;
    line-height: 1;
}

.button--link {
    background: none;
}

.button--secondary {
    background: $google-red;
}

.button--login {
    background: $light-grey;
    color: $dark-grey;
}

.button--login-google-sign-in-img {
    width: 20px;
    margin: 0 10px 0 0;
    vertical-align: middle;
}

.button-container{
    margin: 0;
    padding: 0;
    display: flex;
    justify-content: center;
    width: 100%;
    text-align: center;
    .button {
        width: 35%;
        @media (max-width: $desktop-breakpoint) {
            width: 100%;
        }
    }
}