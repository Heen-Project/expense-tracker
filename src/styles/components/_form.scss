.form {
    display: flex;
    flex-direction: column;
    > * {
        margin-bottom: $l-size;
    }
}

.form__error {
    margin: 0 0 $m-size 0;
    font-style: italic;
    color: $google-red;
}

.SingleDatePicker{
    display: flex;
    justify-content: center;
    align-items: center;
    > div {
        width: 100%;
        .SingleDatePickerInput {
            width: 100%;
            .DateInput {
                width: 100%;
            }
        }
    }
 }

 .form-body {
    margin-bottom: $m-size;
    @media (min-width: $desktop-breakpoint) {
        margin-bottom: $l-size;
    }
}