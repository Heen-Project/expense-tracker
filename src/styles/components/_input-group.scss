.input-group{
    display: flex;
    flex-direction: column;
    margin-bottom: $m-size;
    justify-content: center;
    @media (min-width: $desktop-breakpoint) {
        flex-direction: row;
        margin-bottom: $l-size;
    }
}

.input-group__item{
    margin-bottom: $s-size;
    > * {
        width: 100%;
    }
    @media (min-width: $desktop-breakpoint) {
        margin: 0 $s-size 0 0;
    }
}

.input-group__item:first-child{
    flex-grow: 1;
}

@media (max-width: $desktop-breakpoint) {
    .DateRangePicker{
        display: flex;
        justify-content: center;
        align-items: center;
        > div {
            width: 100%;
            .DateRangePickerInput {
                width: 100%;
                .DateInput {
                    width: 45%;
                }
                .DateRangePickerInput_arrow {
                    width: 5%;
                }
                .DateRangePickerInput_clearDates{
                    @extend .DateRangePickerInput_arrow
                }
            }
        }
     }
}