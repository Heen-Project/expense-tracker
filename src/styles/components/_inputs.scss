.text-input {
    border: 1px solid $box-border;
    height: 50px;
    padding: $s-size;
    margin: 0;
    font-size: $font-size-large;
    font-weight: 300;
}

.select {
    @extend .text-input;
}

.textarea {
    @extend .text-input;
    height: 10rem;
}