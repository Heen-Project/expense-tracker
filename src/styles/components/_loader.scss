.loader {
    height: 100vh;
    width: 100vw;
    display: flex;
    align-items: center;
    justify-content: center;
    background: $off-white;
}

.loader__image{
    object-fit: none;
    border-radius: 100%;
    height: 20rem;
    width: 20rem;
}