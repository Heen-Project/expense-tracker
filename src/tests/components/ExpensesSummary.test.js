import React from 'react'
import { shallow } from 'enzyme'
import { ExpensesSummary } from '../../components/ExpensesSummary'

test('render ExpenseSummary with one expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} ExpensesTotal={195}/>)
    expect(wrapper).toMatchSnapshot()
})

test('render ExpenseSummary with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={12} ExpensesTotal={11295}/>)
    expect(wrapper).toMatchSnapshot()
})