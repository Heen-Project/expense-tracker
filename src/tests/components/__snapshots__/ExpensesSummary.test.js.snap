// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ExpenseSummary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $0.00
     
  </h1>
</div>
`;

exports[`render ExpenseSummary with one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $0.00
     
  </h1>
</div>
`;
